madgraph/interface/reweight_interface.py

```diff
@@ -708,6 +760,235 @@ def do_launch(self, line):
+                E_1_new = math.sqrt(new_event[0].mass**2 + new_event[0].px**2 + new_event[0].py**2 + pz_1_new **2)
+                E_2_new = math.sqrt(new_event[1].mass**2 + new_event[1].px**2 + new_event[1].py**2 + pz_2_new **2)
+                new_event[0].set_momentum(lhe_parser.FourMomentum([E_1_new,new_event[0].px,new_event[0].py,pz_1_new]))
+                new_event[1].set_momentum(lhe_parser.FourMomentum([E_2_new,new_event[1].px,new_event[1].py,pz_2_new]))
+                new_event.pop(to_remove)
+                
+            if fks_i > 1: # final-state recoil
+
+                # Re-scale the energy of fks_i to make it on-shell
+                for ip,part in enumerate(new_event):
+                    if (ip == fks_i):
+                        part.E = math.sqrt(part.mass**2 + part.px**2 + part.py**2 + part.pz**2)
+                        new_E = part.E
+
+                # Find the overall energy in the final state
+                new_E = 0.0
+                for ip,part in enumerate(new_event):
+                    if (ip != fks_j and ip >= 2):
+                        new_E = new_E + part.E
+                
+                # Use one of the initial states to absorb the energy change in the final state
+                new_event[1].set_momentum(lhe_parser.FourMomentum([new_E-new_event[0].E,new_event[1].px,new_event[1].py,new_event[1].pz]))
+                
+                # Change the initial state pz and E
+                pz_1_new = self.recoil_eq(new_event[0],new_event[1])
+                pz_2_new = new_event[0].pz + new_event[1].pz - pz_1_new
+                E_1_new = math.sqrt(new_event[0].mass**2 + new_event[0].px**2 + new_event[0].py**2 + pz_1_new **2)
+                E_2_new = math.sqrt(new_event[1].mass**2 + new_event[1].px**2 + new_event[1].py**2 + pz_2_new **2)
+                new_event[0].set_momentum(lhe_parser.FourMomentum([E_1_new,new_event[0].px,new_event[0].py,pz_1_new]))
+                new_event[1].set_momentum(lhe_parser.FourMomentum([E_2_new,new_event[1].px,new_event[1].py,pz_2_new]))
+                new_event.pop(to_remove)
+            
+        elif fks_type and not recoil:        
+            ## Do it in a more FKS-style
+            if (i == moth[0].get('number')-1):
+                fks_i = i
+                fks_j = j
+            elif (j == moth[0].get('number')-1):
+                fks_i = j
+                fks_j = i
+            to_remove = fks_j
+            new_event = copy.copy(event)
+
+            if fks_i <= 1: # initial-state recoil
+
+                # First boost to partonic CM frame
+                q = lhe_parser.FourMomentum([new_event[0].E+new_event[1].E,new_event[0].px+new_event[1].px,\
+                            new_event[0].py+new_event[1].py,new_event[0].pz+new_event[1].pz])
+                for ip,part in enumerate(new_event):
+                    vec = lhe_parser.FourMomentum([part.E,part.px,part.py,part.pz])
```

Comment:
```suggestion
                    vec = lhe_parser.FourMomentum(part)
```

many similar change of the type can be done continuing downwards. (but github interface is boring to do them all)